src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Yeah, I think we should either prevent (somehow) the swapping of two `Node`s where one is in a sub-tree of the other, or we should put this off until we can come up with a reasonable behavior for that scenario (maybe looking at what other tree libraries do if they handle that kind of case?).

In addition though, I think it would be nice to have a version of `swap` that doesn't carry children with it to be honest.  It would be like swapping the data in two different `Node`s except that the `NodeId`s that refer to them would follow them to their new locations in the tree (meaning parent and child references would need to be updated of course).

Also, do you mind fixing the `` ` `` above?  I think it's preventing this doc test from running as I don't see it in here: 
![image](https://cloud.githubusercontent.com/assets/6172641/21241862/46c34010-c2d7-11e6-93ab-2eb35d416a25.png)